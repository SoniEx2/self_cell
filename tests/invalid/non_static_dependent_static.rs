use once_self_cell::sync::OnceSelfCell;

struct Ast<'a>(&'a str);

fn non_static_dependent_static<'a>() {
    let _: OnceSelfCell<String, Ast<'a>> = OnceSelfCell::new("33".into(), |_| panic!());
}

fn main() {
    non_static_dependent_static();
}
