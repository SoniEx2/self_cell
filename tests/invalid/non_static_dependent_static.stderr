error[E0477]: the type `Ast<'a>` does not fulfill the required lifetime
 --> $DIR/non_static_dependent_static.rs:6:12
  |
6 |     let _: OnceSelfCell<String, Ast<'a>> = OnceSelfCell::new("33".into());
  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: type must satisfy the static lifetime
